test/framework/features/features.gen.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+	
+//WARNING: THIS IS AN AUTO-GENERATED FILE, DO NOT EDIT.
```

Comment:
The project seems to love having yaml -> golang code generation. I personally think its a really bad idea. This entire PR could be 30 lines of code (this file), but is instead 10x as large. If I want to add a feature, I now need to find the yaml file, run `make gen` (this takes ~5 minutes), find out what the generated constant name is, and then I can finally get started. If we just make this go code, the process is dead simple - I just add a string constant and I am good to go.

Additionally, this file is not very useful for developers - I have no clone what `security_mTLS_onbydefault` means, because there are no comments. We can add comments to the yaml file, but then if I want to find the comment I need to try to hunt down the yaml file. If this was just plain go code, I can just jump to the source in my IDE and easily see the comment. You can also write code generation to add the yaml comments to the go code, but please don't - we wasted 100s of hours trying to do this in the past and it still didn't work.

I would suggest we just write this file directly. If we don't like the minor duplication here, we can use https://godoc.org/golang.org/x/tools/cmd/stringer instead maybe?

Reply:
@howardjohn, stringer does not solve the primary duplication problem, which is the duplication between constant names.  For instance, the Networking category should be used hundreds of times across tests, each in different keys.  If this were implemented directly in go, it would be easy to miss a typo like this:

```
const (
  networking_loadbalancing_leastcost
  networking_circuitbraker
  netwroking_mirroring_tofile
)
```

stringer would not fix this typo, and the consequence will be reports that are permanently  altered to include a top level "netwroking" category.  

If `make gen` is a limitation, we should investigate why it is taking so long and fix it, or break frequently run commands into their own subcommands (such as make gen-go).

Reply:
@therealmitchconnors I think I just disagree with you that that duplication is bad, which I think is 100% personal preference. We can always have a test to check for typos. Making `make gen` fast does not solve many problem I mentioned (especially "this is a major time sink")